aspnetcore-webapi/RateLimitingDotNET8/RateLimitingDotNET8.Tests/CustomerControllerTestsConcurrency.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using System.Net;
+
+namespace RateLimitingDotNET8.Tests;
+
+public class CustomerControllerTestsConcurrency : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly HttpClient _client;
+
+    public CustomerControllerTestsConcurrency(WebApplicationFactory<Program> factory)
+    {
+        _client = factory.WithWebHostBuilder(builder =>
+        {
+            builder.ConfigureTestServices(services =>
+            {
+                services.AddAuthentication(defaultScheme: "TestScheme")
+                    .AddScheme<AuthenticationSchemeOptions, TestAuthHandler>(
+                        "TestScheme", options => { });
+
+            });
+        })
+        .CreateClient(new WebApplicationFactoryClientOptions
+        {
+            AllowAutoRedirect = false,
+        });
+    }
+
+    [Fact]
+    public async Task WhenConcurrencyRateLimitedEndpoint_ThenCannotHaveMoreThan10ConcurrentRequests()
+    {
+        var tasks = new List<Task<HttpResponseMessage>>();
+
+        for (int i = 0; i < 12; i++)
```

Comment:
This build was failed because of this test case. When I tested this in local, I found that it is not working as I expected. Currently the QueueLimit is 0. I changed it to 5, so I was expecting any excess request more then 15 will result in 429 "Too many requests", response as you explained in the article. But it is not behaving as described. Please recheck.

Reply:
So, now I left it at 11, and the test isn't failng. Let me know what is happening on your machine?

Reply:
The test case is still failing, you can see the jenkin output log. After commiting the code, please ensure jenkin pipeline is successful.

Reply:
Hi, I tried to max the number, but somehow it always fails on jenkins :/. I have no idea why, it passes locally... I'm at loss at why is this happening. Do you have any ideas, or could I just comment out the test. It may be the issues with resources in jenkins.

Reply:
Hi Dona,

I don't have access to my laptop till Sunday, I will check this on Monday and confirm you.

Thanks

Reply:
Please check Jeff's comment in the article

Reply:
You have not implemented the solution suggested by Jeff, he has explained in detail with screenshot, are you able to see his comments in the article?